tests/analysis/test_lib.py

```diff
@@ -36,32 +35,51 @@ def _AnalyzeEvents(
     session = sessions.Session()
     storage_writer = fake_writer.FakeStorageWriter(session)
     storage_writer.Open()
-    for event in event_objects:
+    for event, event_data in test_events:
+      storage_writer.AddEventData(event_data)
+
+      event.SetEventDataIdentifier(event_data.GetIdentifier())
       storage_writer.AddEvent(event)
 
     mediator = analysis_mediator.AnalysisMediator(
         storage_writer, knowledge_base_object)
 
-    for event in event_objects:
-      plugin.ExamineEvent(mediator, event)
+    for event, event_data in test_events:
+      plugin.ExamineEvent(mediator, event, event_data)
 
     analysis_report = plugin.CompileReport(mediator)
     storage_writer.AddAnalysisReport(analysis_report)
 
     return storage_writer
 
-  def _CreateTestEventObject(self, event_dictionary):
-    """Create a test event with a set of attributes.
+  def _CreateTestEvent(self, event_values):
```

Comment:
This needs some refactoring, as there's a bunch of attributes that are set on the event_data that aren't set in normal operation - see the notes about filename and parser above.

Reply:
ack, I opt to this as part of https://github.com/log2timeline/plaso/issues/2524